features/netsocket/TCPSocket.cpp

```diff
@@ -188,9 +190,8 @@ nsapi_size_or_error_t TCPSocket::recv(void *data, nsapi_size_t size)
 
 void TCPSocket::event()
 {
-    _write_sem.release();
-    _read_sem.release();
-
+    _event_flag.set(WRITE_FLAG);
```

Comment:
Why not to set it in one go? `WRITE_FLAG | READ_FLAG`?

Reply:
Done